internal/executor/shell.go

```diff
@@ -73,12 +73,26 @@ func ShellCommandsAndWait(ctx context.Context, scripts []string, custom_env *map
 
 		return cmd, TimeOutError
 	case <-done:
-		_ = sc.kill()
-
-		if err := sc.piper.Close(ctx); err != nil {
-			handler([]byte(fmt.Sprintf("\nShell session I/O error: %s", err)))
+		// Try to close the piper
+		subCtx, subCtxCancel := context.WithTimeout(ctx, time.Second)
```

Comment:
It does seem like a hack with the one second wait. It seem the core of the problem is `sc.kill()` and that we started to make sure the whole process tree is killed. Which is great for persistence workers and overall cleanliness but might not be as useful for ephemeral environments.

Maybe we should just add a behavioral environment variable like `CIRRUS_SKIP_EXPLICIT_PROCESS_KILLING`?

Reply:
Yeah, it's kind of a hack in an attempt to avoid any changes to the user's workflows.

I've introduced `CIRRUS_ESCAPING_PROCESSES` in https://github.com/cirruslabs/cirrus-ci-agent/pull/220/commits/4875012fb5a4205a6662165ba9e539b97ece8bbd.